src/config.rs

```diff
@@ -113,28 +158,29 @@ impl Config {
     }
 
     pub fn get_instantiated_rules(&self) -> Vec<InstantiatedRule> {
-        let active_rules = self.get_active_rules();
-        if active_rules.is_empty() {
+        let active_rules_and_associated_plugins = self.get_active_rules_and_associated_plugins();
+        if active_rules_and_associated_plugins.is_empty() {
             panic!("No configured active rules");
         }
-        let active_rules = self.filter_based_on_rule_argument(active_rules);
-        let instantiated_rules = active_rules
+        let active_rules_and_associated_plugins =
+            self.filter_based_on_rule_argument(active_rules_and_associated_plugins);
+        active_rules_and_associated_plugins
             .into_iter()
-            .map(|rule| InstantiatedRule::new(rule.clone(), self))
-            .collect::<Vec<_>>();
-        if instantiated_rules.is_empty() {
-            panic!("Invalid rule name: {:?}", self.rule.as_ref().unwrap());
-        }
-        instantiated_rules
+            .map(|(rule, plugin_index)| InstantiatedRule::new(rule.clone(), plugin_index, self))
```

Comment:
Now this weird `InstantiatedRule` "wrapper" type is sort of becoming more useful, it's where we're storing/exposing the association between a rule and which plugin it "came from"